src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -2377,22 +2376,17 @@ def _where_construct_handler(self, node, parent):
         for name_obj in name_list:
             name = str(name_obj)
             if name not in symbol_table:
-                # TODO #630 some of these symbols will be put in the wrong
-                # symbol table. We need support for creating a unique symbol
+                # We need support for creating a unique symbol
```

Comment:
I hadn't updated the description correctly. On looking at the code again I have decided to remove most of it. What it is doing is adding in all symbols it can find to the local symbol table in order to make sure that any symbols used only within a code block do not cause a name clash with the loop variables that are about to be declared. Whilst this makes sense, the implementation does not. I think the code block logic should add any symbols contained within it to the symbol table. Any addition of symbols would then conform to the symbol table hierarchy. I've created issue #831 which captures the problem.